mobile/Ihc/services/LocalDataService.js

```diff
@@ -302,6 +363,68 @@ export function markPatientsUploaded() {
   });
 }
 
+export function markMedicationsUploaded() {
+  const medications = Object.values(realm.objects('Medication').filtered('needToUpload = true'));
+  realm.write(() => {
+    medications.forEach(medication => {
+      medication.needToUpload = false;
+    });
+  });
+}
+
+export function handleDownloadedMedications(medications) {
+  const fails = new Set();
+
+  medications.forEach( incomingMedication => {
+    const existingMedication = realm.objects('Medication')
+      .filtered('key = "' + incomingMedication.key + '"')['0'];
+
+    // Medication received does not exist yet
+    if(!existingMedication) {
+      realm.write(() => {
+        realm.create('Medication', incomingMedication);
+      });
+    }
+    else {
+      if (incomingMedication.lastUpdated < existingMedication.lastUpdated) {
+        throw new Error('Received a medication that is out-of-date. Did you upload updates yet?');
+      }
+
+      // The case where MedicationInventoryScreen is refreshing
+      if (incomingMedication.lastUpdated == existingMedication.lastUpdated) {
+        return;
+      }
+
+      // Actually update the medication
+      if(!updateMedication(existingMedication.key, incomingMedication))
+        fails.add(existingMedication.key);
+
+      // Update that medication's updated timestamp
+      realm.write(() => {
+        // If medication finished updating successfully
+        if(!fails.has(incomingMedication.key)) {
```

Comment:
Really small change, but can we switch line 405 and 403? We can avoid entering the realm.write() call if the condition isn't met